onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -16,47 +18,50 @@
 #include "core/framework/mlvalue_name_idx_map.h"
 #include "core/framework/sequential_execution_plan.h"
 #include "core/framework/session_state.h"
-#include "core/framework/tensorutils.h"
 #include "core/framework/tensorprotoutils.h"
 #include "core/framework/utils.h"
+#include "core/framework/mem_buffer.h"
 
 namespace onnxruntime {
 
 static common::Status SaveMLValueNameIndexMapping(const GraphViewer& graph_viewer,
                                                   MLValueNameIdxMap& mlvalue_name_idx_map,
                                                   const logging::Logger& logger);
 
-using SaveTensorFunc = std::function<void(int idx, const onnxruntime::MLValue&)>;
-
-static common::Status SaveInitializedTensors(const onnxruntime::Graph& graph,
-                                             bool enable_memory_pattern,
+// T should have signature of '(int idx, const onnxruntime::MLValue& value, const OrtDeleter& d) -> Status'
+template <typename T>
+static common::Status SaveInitializedTensors(const Env& env, const std::basic_string<PATH_CHAR_TYPE>& graph_loc,
+                                             const onnxruntime::Graph& graph,
                                              const SequentialExecutionPlan& execution_plan,
                                              const ExecutionProviders& exec_providers,
                                              const MLValueNameIdxMap& mlvalue_name_idx_map,
                                              std::map<OrtAllocatorInfo, BufferUniquePtr>& weights_buffers,
-                                             const SaveTensorFunc& save_tensor_func,
-                                             const logging::Logger& logger);
+                                             const T& save_tensor_func, const logging::Logger& logger);
 
 static common::Status SaveKernels(const ExecutionProviders& execution_providers,
                                   SessionState& session_state,
                                   const KernelRegistryManager& custom_registry_manager,
                                   const logging::Logger& logger);
 
+/**
+ *
+ * \param implicit_inputs Could be NULL
+ */
 static common::Status SaveInputOutputNamesToNodeMapping(const onnxruntime::Graph& graph,
                                                         const KernelRegistryManager& custom_registry_manager,
                                                         SessionState& session_state,
                                                         const std::vector<NodeArg*>* implicit_inputs);
 
-SessionStateInitializer::SessionStateInitializer(onnxruntime::Graph& graph,
-                                                 SessionState& session_state,
+SessionStateInitializer::SessionStateInitializer(const std::basic_string<PATH_CHAR_TYPE>& graph_loc,
```

Comment:
>graph_loc [](start = 90, length = 9)

This is the directory the model/graph was loaded from isn't it? Could we use a more explicit name to clarify?

Reply:
No. The file path. not dir path.

---
In reply to: [261364866](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261364866) [](ancestors = 261364866)